--- pygobject-2.15.3.orig/configure.ac	2008-09-03 18:43:14.000000000 +0200
+++ pygobject-2.15.3/configure.ac	2008-09-03 18:43:17.000000000 +0200
@@ -15,6 +15,7 @@
 AC_INIT(pygobject, pygobject_version,
         [http://bugzilla.gnome.org/enter_bug.cgi?product=pygobject])
 AC_SUBST(ACLOCAL_AMFLAGS, "-I m4 -I .")
+AC_CONFIG_MACRO_DIR([m4])
 
 AC_DEFINE(PYGOBJECT_MAJOR_VERSION, pygobject_major_version, [pygobject major version])
 AC_SUBST(PYGOBJECT_MAJOR_VERSION, pygobject_major_version)
